include/dlaf/util_matrix.h

```diff
@@ -146,5 +154,207 @@ void assertMultipliableMatrices(const MatrixConst& mat_a, const Matrix& mat_b, b
+  const matrix::Distribution& dist = matrix.distribution();
+
+  // Check if matrix is square
+  util_matrix::assertSizeSquare(matrix, "set_hermitian_random_positive_definite", "matrix");
+  // Check if block matrix is square
+  util_matrix::assertBlocksizeSquare(matrix, "set_hermitian_random_positive_definite", "matrix");
+
+  std::size_t offset_value = 2 * std::max(matrix.size().rows(), matrix.size().cols());
+
+  for (SizeType tile_j = 0; tile_j < dist.localNrTiles().cols(); ++tile_j) {
+    for (SizeType tile_i = 0; tile_i < dist.localNrTiles().rows(); ++tile_i) {
+      LocalTileIndex tile_wrt_local{tile_i, tile_j};
+      GlobalTileIndex tile_wrt_global = dist.globalTileIndex(tile_wrt_local);
+
+      auto tl_index = dist.globalElementIndex(tile_wrt_global, {0, 0});
+
+      auto tile_position = dlaf::common::position(tile_wrt_global);
+
+      // compute the same seed for original and "transposed" tiles, so transposed ones will know the
+      // values of the original one without the need of accessing real values (nor communication in case
+      // of distributed matrices)
+      size_t seed;
+      if (tile_position <= dlaf::common::Position::DIAGONAL)
+        seed = tl_index.row() * matrix.size().cols() + tl_index.col();
+      else
+        seed = tl_index.col() * matrix.size().rows() + tl_index.row();
+
+      hpx::dataflow(
+          hpx::util::unwrapping([tile_position, tl_index, seed, offset_value](auto&& tile) {
+            internal::getter_random_hermitian_positive_definite<T> value_at(offset_value, seed);
+
+            if (tile_position == dlaf::common::Position::DIAGONAL) {
+              // for diagonal tiles get just lower matrix values and set value for both straight and
+              // transposed indices
+              for (SizeType j = 0; j < tile.size().cols(); ++j) {
+                for (SizeType i = 0; i <= j; ++i) {
+                  auto value = value_at(GlobalElementIndex{tl_index.row() + i, tl_index.col() + j});
+
+                  tile(TileElementIndex{i, j}) = value;
+                  if (i != j)
+                    tile(TileElementIndex{j, i}) = dlaf::conj(value);
+                }
+              }
+            }
+            else {
+              // random values are requested in the same order for both original and transposed
+              for (SizeType j = 0; j < tile.size().cols(); ++j) {
+                for (SizeType i = 0; i < tile.size().rows(); ++i) {
+                  auto value = value_at(GlobalElementIndex{tl_index.row() + i, tl_index.col() + j});
+
+                  // but they are set row-wise in the original tile and col-wise in the transposed one
+                  if (tile_position == dlaf::common::Position::LOWER)
+                    tile(TileElementIndex{i, j}) = value;
+                  else
+                    tile(TileElementIndex{j, i}) = value;
+                }
+              }
+            }
+          }),
+          matrix(tile_wrt_local));
+    }
+  }
```

Comment:
Why you check again the position in the random getter?
(See internal comments)

Moreover this approach may lead to think that value_at({i, j}) gives always the same value for a given (i, j)